docs/graphs/classdiagram_lib.dot

```diff
@@ -0,0 +1,34 @@
+digraph hierarchy {
+size="5,5"
+node[   shape=record,
+        style=filled,
+        fillcolor=gray95]
+edge[dir=back, arrowtail=none]
+
+
+ps[label = "{ProgramState|
+                all_transactions\l
+                filtered_transactions\l|
+                + setup()\l
+                + add(dataframe)\l
+                + apply(filter)\l
+                + categorize(row, category)\l
+                + store()\l
+                }"]
+imp[label = "{TextImporter|| + read(file) : dataframe}"]
+tfd[label = "{TextfileDescription|
+                columns\l
+                dialect\l|
+                }"]
+db[label = "{DatabaseInterface|
+                db_credentials\l|
+                + load_all_transactions() : dataframe\l
+                + store_transactions(dataframe)\l
+                + load_iban_lookup() : dataframe\l}"]
```

Comment:
I like `DatabaseInterface`, but I think that it mixes different kinds of databases: transaction databases on the one hand and "IBAN-related data" databases on the other. I think these different concerns should be split into different interfaces/classes since this will make handling those interfaces/classes more easy and will make them more testable.

I propose splitting `DatabaseInterface` into two interfaces like `TransactionDatabaseInterface` (containing `load_all_transactions` and `store_transactions`) and something like `AccountInformationDatabaseInterface` (bad name yet, purpose not very clear yet).